### PR TITLE
Add additional conditions in queryset filter to filter outdated tasks

### PR DESCRIPTION
**Pull request by MichalW-CP**
Resolves https://github.com/golemfactory/concent/issues/668

### CODE REVIEW

File: concent_api/core/admin.py [hunk 1]
```diff
@@ -18,22 +18,25 @@ class ActivePassiveDownloadsStateFilter(admin.SimpleListFilter):
...
+            ('passive', 'Passive and timed out'),
+            ('active_or_downloads', 'Active and not timed out or with active downloads')
         )
 
     def queryset(self, request, queryset: QuerySet) -> QuerySet:
         assert self.value() in {'active', 'passive', 'active_or_downloads', None}
+        current_timestamp = get_current_utc_timestamp()
         if self.value() == 'active':
-            return queryset.filter(state__in=ACTIVE_STATE_NAMES)
+            return queryset.filter(Q(state__in=ACTIVE_STATE_NAMES, next_deadline__gte=parse_timestamp_to_utc_datetime(current_timestamp)))
```
Comment (rwrzesien) [inline concent_api/core/admin.py]: `Q` is not needed here, you can use several conditions in `.filter` unless you want to make them sql `OR` (`|`), and they will work as sql `AND` (this is Django default).
Please note that in `.exclude` it works differently.

File: concent_api/core/admin.py [hunk 2]
```diff
@@ -18,22 +18,25 @@ class ActivePassiveDownloadsStateFilter(admin.SimpleListFilter):
...
+            return queryset.filter(
+                Q(state__in=PASSIVE_STATE_NAMES) |
+                Q(state__in=ACTIVE_STATE_NAMES, next_deadline__lte=parse_timestamp_to_utc_datetime(current_timestamp))
+            )
         elif self.value() == 'active_or_downloads':
-            current_timestamp = get_current_utc_timestamp()
             return Subtask.objects_with_timing_columns.filter(
                 Q(download_deadline__gte=current_timestamp, state=Subtask.SubtaskState.RESULT_UPLOADED.name) |  # pylint: disable=no-member
-                Q(state__in=ACTIVE_STATE_NAMES)
+                Q(state__in=ACTIVE_STATE_NAMES, next_deadline__gte=parse_timestamp_to_utc_datetime(current_timestamp))
```
Comment (rwrzesien) [inline concent_api/core/admin.py]: Same here.

File: concent_api/core/admin.py [hunk 3]
```diff
@@ -18,22 +18,25 @@ class ActivePassiveDownloadsStateFilter(admin.SimpleListFilter):
...
         assert self.value() in {'active', 'passive', 'active_or_downloads', None}
+        current_timestamp = get_current_utc_timestamp()
         if self.value() == 'active':
-            return queryset.filter(state__in=ACTIVE_STATE_NAMES)
+            return queryset.filter(Q(state__in=ACTIVE_STATE_NAMES, next_deadline__gte=parse_timestamp_to_utc_datetime(current_timestamp)))
         elif self.value() == 'passive':
-            return queryset.filter(state__in=PASSIVE_STATE_NAMES)
+            return queryset.filter(
+                Q(state__in=PASSIVE_STATE_NAMES) |
+                Q(state__in=ACTIVE_STATE_NAMES, next_deadline__lte=parse_timestamp_to_utc_datetime(current_timestamp))
```
Comment (rwrzesien) [inline concent_api/core/admin.py]: You are treating equal values as both timed out and not (this query and two others). It should be either way.

File: concent_api/core/admin.py [hunk 4]
```diff
@@ -18,22 +18,25 @@ class ActivePassiveDownloadsStateFilter(admin.SimpleListFilter):
 
     def lookups(self, request, model_admin):
         return (
-            ('active', 'Active'),
-            ('passive', 'Passive'),
-            ('active_or_downloads', 'Active or with active downloads')
+            ('active', 'Active and not timed out '),
```
Comment (rwrzesien) [inline concent_api/core/admin.py]: I would rephrase it a little bit to make it more clear and match what is in querysets:
"Not timed out active"
"Passive or timed out active"
"Not timed out active or active with pending downloads"

File: concent_api/core/admin.py (hunk 1 above)
Comment (MichalW-CP) [reply concent_api/core/admin.py]: Done.

File: concent_api/core/admin.py (hunk 3 above)
Comment (MichalW-CP) [reply concent_api/core/admin.py]: Done.

File: concent_api/core/admin.py (hunk 4 above)
Comment (MichalW-CP) [reply concent_api/core/admin.py]: Done.

File: concent_api/core/admin.py [hunk 5]
```diff
@@ -18,22 +18,25 @@ class ActivePassiveDownloadsStateFilter(admin.SimpleListFilter):
...
+            ('passive', 'Passive or timed out active'),
+            ('active_or_downloads', 'Not timed out active or active with pending downloads')
         )
 
     def queryset(self, request, queryset: QuerySet) -> QuerySet:
         assert self.value() in {'active', 'passive', 'active_or_downloads', None}
+        current_timestamp = get_current_utc_timestamp()
         if self.value() == 'active':
-            return queryset.filter(state__in=ACTIVE_STATE_NAMES)
+            return queryset.filter(Q(state__in=ACTIVE_STATE_NAMES) & Q(next_deadline__gte=parse_timestamp_to_utc_datetime(current_timestamp)))
```
Comment (rwrzesien) [inline concent_api/core/admin.py]: But you don't need `Q`s and `&` here, it will work this way if you just do:
```
queryset.filter(
    state__in=ACTIVE_STATE_NAMES,
    next_deadline__gte=parse_timestamp_to_utc_datetime(current_timestamp)
)
```

It looks simpler.

**Comment by cameel** [comment]
Please use more concrete names for your branches and commits. Instead of "additional conditions" it would be more useful to say which conditions you mean.

File: concent_api/core/admin.py [hunk 6]
```diff
@@ -18,22 +18,25 @@ class ActivePassiveDownloadsStateFilter(admin.SimpleListFilter):
 
     def lookups(self, request, model_admin):
         return (
-            ('active', 'Active'),
-            ('passive', 'Passive'),
-            ('active_or_downloads', 'Active or with active downloads')
+            ('active', 'Not timed out active'),
```
Comment (dybi) [inline concent_api/core/admin.py]: As discussed - I do not like the new names - they complicate otherwise simple things

File: concent_api/core/admin.py [hunk 7]
```diff
@@ -18,22 +18,25 @@ class ActivePassiveDownloadsStateFilter(admin.SimpleListFilter):
...
+                Q(state__in=PASSIVE_STATE_NAMES) |
+                Q(state__in=ACTIVE_STATE_NAMES) & Q(next_deadline__lte=parse_timestamp_to_utc_datetime(current_timestamp))
+            )
         elif self.value() == 'active_or_downloads':
-            current_timestamp = get_current_utc_timestamp()
             return Subtask.objects_with_timing_columns.filter(
-                Q(download_deadline__gte=current_timestamp, state=Subtask.SubtaskState.RESULT_UPLOADED.name) |  # pylint: disable=no-member
-                Q(state__in=ACTIVE_STATE_NAMES)
+                Q(download_deadline__gte=current_timestamp) & Q(state=Subtask.SubtaskState.RESULT_UPLOADED.name) |  # pylint: disable=no-member
+                Q(state__in=ACTIVE_STATE_NAMES) & Q(next_deadline__gte=parse_timestamp_to_utc_datetime(current_timestamp))
```
[the remark this answers is not in the document]
Comment (MichalW-CP) [reply concent_api/core/admin.py]: Done.

File: concent_api/core/admin.py (hunk 5 above)
Comment (MichalW-CP) [reply concent_api/core/admin.py]: Done.

File: concent_api/core/admin.py (hunk 3 above)
Comment (MichalW-CP) [reply concent_api/core/admin.py]: Done.

**Comment by cameel** [comment]
@MichalW-CP So what about the names?

**Comment by MichalW-CP** [comment]
@cameel Do you mean commit name or filter names?
